File: lessons/compiled_lang/go/src/m10.function.go
package main

import (
	"fmt"
	"strings"
)

func capitalize(s string) string {
	return strings.ToUpper(s)
}

func main() {
	s := "ibm"
	fmt.Printf("The current string is: \"%s\".\n", s)

	result := capitalize(s)
	fmt.Printf("The capitalized string is: \"%s\".\n", result)
}

File: lessons/compiled_lang/cpp/src/m10.function.cpp
#include <algorithm>
#include <iostream>
#include <string>

std::string capitalize(std::string s) {
    std::transform(s.begin(), s.end(), s.begin(), ::toupper);
    return s;
}

int main() {
    std::string s = "ibm";
    std::cout << "The current string is: \"" << s << "\"." << std::endl;

    std::string result = capitalize(s);
    std::cout << "The capitalized string is: \"" << result << "\"." << std::endl;

    return 0;
}
